examples/tutorials/cnndm_summarization.py

```diff
@@ -0,0 +1,309 @@
+"""
+CNNDM Text Summarization with T5-Base model
+=======================================================
+
+**Author**: `Pendo Abbo <pabbo@fb.com>`__
+
+"""
+
+######################################################################
+# Overview
+# --------
+#
+# This tutorial demonstrates how to use a pre-trained T5 Model for text summarization on the CNN-DailyMail dataset.
+# We will demonstrate how to use the torchtext library to:
+#
+# 1. Build a text pre-processing pipeline for a T5 model
+# 2. Read in the CNNDM dataset and pre-process the text
+# 3. Instantiate a pre-trained T5 model with base configuration, and perform text summarization on input text
+#
+#
+
+######################################################################
+# Common imports
+# --------------
+import torch
+import torch.nn.functional as F
+
+DEVICE = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+
+
+#######################################################################
+# Data Transformation
+# -------------------
+#
+# The T5 model does not work with raw text. Instead, it requires the text to be transformed into numerical form
+# in order to perform training and inference. The following transformations are required for the T5 model:
+#
+# 1. Tokenize text
+# 2. Convert tokens into (integer) IDs
+# 3. Truncate the sequences to a specified maximum length
+# 4. Add end-of-sequence (EOS) and padding token IDs
+#
+# T5 uses a SentencePiece model for text tokenization. Below, we use a pre-trained SentencePiece model to build
+# the text pre-processing pipeline using torchtext's T5Transform. Note that the transform supports both
+# batched and non-batched text input (i.e. one can either pass a single sentence or a list of sentences), however
+# the T5 model expects the input to be batched.
+#
+
+from torchtext.prototype.models import T5Transform
+
+padding_idx = 0
+eos_idx = 1
+max_seq_len = 512
+t5_sp_model_path = r"https://download.pytorch.org/models/text/t5_tokenizer_base.model"
```

Comment:
nit: I don't think we need to treat this URL as a raw string since there's no backslashes present